viper/types.py

```diff
@@ -311,19 +317,20 @@ def ceil32(x):
 
 
 # Gets the number of memory or storage keys needed to represent a given type
-def get_size_of_type(typ):
+def get_size_of_type(typ, in_return=False):
     if isinstance(typ, BaseType):
         return 1
     elif isinstance(typ, ByteArrayType):
-        return ceil32(typ.maxlen) // 32 + 2
+        return_offset = 1 if in_return else 0
+        return ceil32(typ.maxlen) // 32 + 2 + return_offset
     elif isinstance(typ, ListType):
```

Comment:
I think you could take out line 324 `return_offset = 1 if in_return else 0` if you want to.

Reply:
@DavidKnott ha, thanks! Good catch - yes this isn't need at all anymore. I needed it in a previos iteration of the work.